### PR TITLE
Add SSRF protection

### PR DESCRIPTION
**Pull request by bitterpanda63**

### CODE REVIEW

File: aikido_firewall/vulnerabilities/ssrf/inspect_getaddrinfo_result.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+"""
+!!!!!!!!!!!!!!!!!1
```
Comment (willem-delbare) [inline aikido_firewall/vulnerabilities/ssrf/inspect_getaddrinfo_result.py]: ?

Comment (bitterpanda63) [reply aikido_firewall/vulnerabilities/ssrf/inspect_getaddrinfo_result.py]: Done

File: aikido_firewall/vulnerabilities/ssrf/inspect_getaddrinfo_result.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+def inspect_getaddrinfo_result(dns_results, hostname, port):
+    """Inspect the results of a getaddrinfo() call"""
+    if not hostname or try_parse_url(hostname) is not None:
+        #  If the hostname is an IP address, we don't need to inspect it
+        logger.debug("Hostname %s is actually an IP address, ignoring", hostname)
+        return
+
+    context = get_current_context()
+
+    # TO DO : Implement checks for "protection off" for a specific route
```
Comment (willem-delbare) [inline aikido_firewall/vulnerabilities/ssrf/inspect_getaddrinfo_result.py]: geen todos

File: aikido_firewall/vulnerabilities/ssrf/inspect_getaddrinfo_result.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+from aikido_firewall.helpers.try_parse_url import try_parse_url
+from aikido_firewall.context import get_current_context
+from aikido_firewall.helpers.logging import logger
+from aikido_firewall.background_process import get_comms
+from aikido_firewall.errors import AikidoSSRF
+from .imds import is_trusted_hostname, is_imds_ip_address
+from .is_private_ip import is_private_ip
+from .find_hostname_in_context import find_hostname_in_context
+
+
```
Comment (willem-delbare) [inline aikido_firewall/vulnerabilities/ssrf/inspect_getaddrinfo_result.py]:
```suggestion
#  gets called when the result of the DNS resolution has come in
```

File: aikido_firewall/vulnerabilities/ssrf/inspect_getaddrinfo_result.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    """Inspect the results of a getaddrinfo() call"""
+    if not hostname or try_parse_url(hostname) is not None:
+        #  If the hostname is an IP address, we don't need to inspect it
+        logger.debug("Hostname %s is actually an IP address, ignoring", hostname)
+        return
+
+    context = get_current_context()
+
+    should_block_res = get_comms().send_data_to_bg_process(
+        action="READ_PROPERTY", obj="block", receive=True
```
Comment (willem-delbare) [inline aikido_firewall/vulnerabilities/ssrf/inspect_getaddrinfo_result.py:26]: only do it when needed

**Pull request merged**